base/show.jl

```diff
@@ -647,7 +648,9 @@ macro show(exs...)
 end
 
 function show(io::IO, tn::Core.TypeName)
+    print(io, "typename(")
     show_type_name(io, tn)
+    print(io, ")")
```

Comment:
Maybe something like `TypeName Array`, so it doesn't look like syntax that might actually work?

Reply:
We actually do have a `typename` function that works this way